server/py/services/api/crud/model_monitoring/model_endpoints.py

```diff
@@ -672,8 +676,19 @@ def get_model_endpoints_metrics(
             raise mlrun.errors.MLRunInvalidArgumentError(
                 "Type must be either 'metric' or 'result'"
             )
-
-        return tsdb_connector.df_to_metrics_list(df=df, type=type, project=project)
+        if metrics_format == mm_constants.GetMetricsFormat.SINGLE:
+            return tsdb_connector.df_to_metrics_list(df=df, type=type, project=project)
+        elif metrics_format == mm_constants.GetMetricsFormat.SEPARATION:
+            return tsdb_connector.df_to_metrics_grouped_dict(
+                df=df, type=type, project=project
+            )
+        elif metrics_format == mm_constants.GetMetricsFormat.INTERSECTION:
+            pass
+            #  TODO create INTERSECTION options
```

Comment:
```suggestion
            # TODO: create INTERSECTION options
```

Reply:
But are we sure we want it to return `None`? Rather than raise an error?

Reply:
Added intersection mode:)